kivy/uix/widget.py

```diff
@@ -660,6 +660,73 @@ def export_to_png(self, filename, *args, **kwargs):
 
         return True
 
+    def export_to_base64(self, *args, **kwargs):
+        '''Saves an image of the widget and its children in base64 format saves into memory instead of storing it.
+        Works by removing the widget canvas from its
+        parent, rendering to an :class:`~kivy.graphics.fbo.Fbo`, and calling
+        :meth:`~kivy.graphics.texture.Texture.save`.
+
+        .. note::
+
+            The image includes only this widget and its children. If you want
+            to include widgets elsewhere in the tree, you must call
+            :meth:`~Widget.export_to_base64` from their common parent, or use
+            :meth:`~kivy.core.window.WindowBase.screenshot` to capture the
+            whole window.
+
+        .. note::
+
+            The image will be returned in a base64 format.
+
+        .. versionadded:: 1.9.0
+
+        :Parameters:
+            `method`: str
+                empty or 'web'. Note: None is for returning the image in Kivy and to be shown.
+            `scale`: float
+                The amount by which to scale the saved image, defaults to 1.
+
+                .. versionadded:: 1.11.0
+        '''
+        try:
+            from PIL import Image
+        except ImportError as error:
+            raise Exception(error)
+        from io import BytesIO
+
+        import base64
```

Comment:
Question: why the lazy imports?

Reply:
You mean why I use it only in the function and not for the entire widget?
If so: I don't think it's useful to call them for all the other widget functions.

Reply:
Yes I mean why importing not in the top of the file? Usually this is where we put imports unless a specific reason, for instance to workaround dependency cycles, but there're other reasons.
Since lazy importing is the exception and not the rule, what I usually do is to put a comment explaining why I would do a lazy import in a specific place, so later maintainers won't move them by mistake unless it's for good reason

Reply:
addressed in daab224